packages/integration/lib/src/bloc/feedroom/feedroom_bloc.dart

```diff
@@ -0,0 +1,288 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+
+import 'package:likeminds_feed_flutter_core/likeminds_feed_core.dart';
+
+part 'feedroom_event.dart';
+part 'feedroom_state.dart';
+
+class LMFeedRoomBloc extends Bloc<LMFeedRoomEvent, LMFeedRoomState> {
+  // list of selected topics by the user
+  List<LMTopicViewData> selectedTopics = [];
+  // list of all the topics
+  Map<String, LMTopicViewData> topics = {};
+// list of all the users
+  Map<String, LMUserViewData> users = {};
+  // list of all the widgets
+  Map<String, LMWidgetViewData> widgets = {};
+
+  static LMFeedRoomBloc? _instance;
+
+  static LMFeedRoomBloc get instance => _instance ??= LMFeedRoomBloc._();
+
+  LMFeedRoomBloc._() : super(LMFeedRoomInitialState()) {
+    on<LMFeedRoomEvent>((event, emit) async {
+      if (event is LMFeedGetFeedRoomEvent) {
+        await _mapLMGetFeedRoomToState(event: event, emit: emit);
+      }
+      if (event is LMFeedGetFeedRoomListEvent) {
+        await _mapLMGetFeedRoomListToState(event: event, emit: emit);
+      }
+    });
+  }
+
+  FutureOr<void> _mapLMGetFeedRoomToState({
+    required LMFeedGetFeedRoomEvent event,
+    required Emitter<LMFeedRoomState> emit,
+  }) async {
+    Map<String, LMUserViewData> users = {};
+    Map<String, LMTopicViewData> topics = {};
+    Map<String, LMWidgetViewData> widgets = {};
+
+    if (state is LMFeedRoomLoadedState) {
+      LMFeedRoomLoadedState prevState = state as LMFeedRoomLoadedState;
+      users = prevState.users;
+      topics = prevState.topics;
+      widgets = prevState.widgets;
+
+      emit(LMFeedRoomPaginationLoadingState(
+        posts: prevState.posts,
+        users: prevState.users,
+        widgets: prevState.widgets,
+        topics: prevState.topics,
+        feedRoom: prevState.feedRoom,
+      ));
+    } else {
+      emit(LMFeedRoomLoadingState());
+    }
+
+    GetFeedOfFeedRoomResponse response =
+        await LMFeedCore.client.getFeedOfFeedRoom(
+      (GetFeedOfFeedRoomRequestBuilder()
+            ..feedroomId(event.feedRoomId)
+            ..topicIds(selectedTopics.map((e) => e.id).toList())
+            ..page(event.offset)
+            ..pageSize(10))
+          .build(),
+    );
+    if (!response.success) {
+      emit(
+        LMFeedRoomErrorState(
+          message: response.errorMessage ?? 'An error occurred',
+        ),
+      );
+    }
+
+    users.addAll(
```

Comment:
we are not mapping widgets received in response?

Also we have to send the widgets in topics and user view data as well.